docs/checkit.js

```diff
@@ -460,6 +460,22 @@ CheckIt.prototype.generateCheckmarks = function(calObj, $div) {
      })
 };
 
+CheckIt.prototype.removeEmptyWeeks = function(calObj, $div) {
+    // Remove empty weeks from the calendar
+        
+    $div.find('.month').find('.week').each( function(index) {
+        var counter = 0;
+        $(this).find('td').each(function(td) {
+            if ($(this).children().hasClass("nill")) 
+            counter +=1;
+        })
+        if (counter === 7) 
+        {
+            $(this).remove();
+        }
+    })
```

Comment:
This is a clever way to figure out if the week should be removed or not, but you can also just ask jquery how many things have the `nil` class.

Try something like `$(this).children('.nil').length`